### PR TITLE
Feature: add primitive mesh demo and update sprite renderer

### PR DESCRIPTION
**Pull request by singlecoder**
- Update sprite renderer demo
- Add mesh renderer demo
- Modify template

Note: For the sprite renderer after refactor

### CODE REVIEW

**Review by eyworldwide — APPROVED** [review]

File: demos/mesh-renderer/index.ts [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+} from "oasis-engine";
+
+class RotateScript extends Script {
+  private _rotateX: number;
+  private _rotateY: number;
+
+  constructor(entity: Entity) {
+    super(entity);
+
+    this._rotateX = 0.5;
```
Comment (GuoLei1990) [inline demos/mesh-renderer/index.ts]: private _rotateX: number=0.5;

File: demos/mesh-renderer/index.ts [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    this._rotateY = 0.6;
+  }
+  onUpdate() {
+    this.entity.transform.rotate(this._rotateX, this._rotateY, 0);
```
Comment (GuoLei1990) [inline demos/mesh-renderer/index.ts]: why not write this.entity.transform.rotate(0.5, 0.6, 0);

File: demos/primitive-mesh/index.ts [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+      const cylinderTransform = cylinderEntity.transform;
+      cylinderTransform.setPosition(posX, -distanceY * 1.5, 0);
+      cylinderTransform.setScale(3, 3, 3);
+      cylinderEntity.addComponent(RotateScript);
+      const cylinderRenderer = cylinderEntity.addComponent(MeshRenderer);
+      const cylinderMaterial = new BlinnPhongMaterial(engine);
+      cylinderMaterial.emissiveTexture = resource;
+      cylinderMaterial.emissiveColor.setValue(1, 1, 1, 1);
+      cylinderRenderer.mesh = PrimitiveMesh.createCylinder(engine, 0.5, 2, (i + 1) * 6, 1);
+      cylinderRenderer.setMaterial(cylinderMaterial);
```
Comment (zhuxudong) [inline demos/primitive-mesh/index.ts:113]: use the same material

File: demos/sprite-renderer/index.ts [hunk 4]
```diff
@@ -1,72 +1,94 @@
...
+    this._scale = 1.0;
+    this._scaleFlag = true;
   }
 
   onUpdate() {
+    // Update position.
     this._curRadian += 0.01;
-    const { _radius, _curRadian } = this;
+    const { _radius, _curRadian, entity } = this;
+    const { transform } = entity;
```
Comment (zhuxudong) [inline demos/sprite-renderer/index.ts:33]: could combine these code to one line.
```
const { _radius, _curRadian, entity:{transform} } = this;
```

File: demos/mesh-renderer/index.ts (hunk 1 above)
Comment (singlecoder) [reply demos/mesh-renderer/index.ts]: > private _rotateX: number=0.5;

done

File: demos/mesh-renderer/index.ts (hunk 2 above)
Comment (singlecoder) [reply demos/mesh-renderer/index.ts]: > why not write this.entity.transform.rotate(0.5, 0.6, 0);

done

File: demos/primitive-mesh/index.ts (hunk 3 above)
Comment (singlecoder) [reply demos/primitive-mesh/index.ts]: > use the same material

done

File: demos/primitive-mesh/index.ts [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+    material.emissiveColor.setValue(1, 1, 1, 1);
+
+    for (let i = 0; i < 3; ++i) {
+      const posX = (i - 1) * distanceX;
+
+      // Create cuboid
+      const cuboidEntity = rootEntity.createChild("cuboid");
+      const cuboidTransform = cuboidEntity.transform;
+      cuboidTransform.setPosition(posX, distanceY * 1.5, 0);
+      cuboidTransform.setScale(3, 3, 3);
```
Comment (GuoLei1990) [inline demos/primitive-mesh/index.ts:66]: can remove setScale and let camera more near

File: demos/primitive-mesh/index.ts [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+
+      // Create sphere
+      const sphereEntity = rootEntity.createChild("sphere");
+      const sphereTransform = sphereEntity.transform;
+      sphereTransform.setPosition(posX, distanceY * 0.5, 0);
+      sphereTransform.setScale(3, 3, 3);
+      sphereEntity.addComponent(RotateScript);
+      const sphereRenderer = sphereEntity.addComponent(MeshRenderer);
+      sphereRenderer.mesh = PrimitiveMesh.createSphere(engine, 0.5, (i + 1) * 6);
+      sphereRenderer.setMaterial(material);
```
Comment (GuoLei1990) [inline demos/primitive-mesh/index.ts:80]: You can Encapsulation with one function and set different postion and mesh with params

File: demos/sprite-renderer/index.ts [hunk 7]
```diff
@@ -1,72 +1,94 @@
...
+// Script for sprite
 class SpriteController extends Script {
+  static _curRotation: number = 0;
   private _curRadian: number;
   private _radius: number;
+  private _scale: number;
+  private _scaleFlag: boolean;
 
   onAwake() {
     this._curRadian = 0;
```
Comment (GuoLei1990) [inline demos/sprite-renderer/index.ts:23]: not write init value in here

Comment (singlecoder) [reply demos/sprite-renderer/index.ts:24]: > not write init value in here

node

File: demos/primitive-mesh/index.ts (hunk 6 above)
Comment (singlecoder) [reply demos/primitive-mesh/index.ts]: > You can Encapsulation with one function and set different postion and mesh with params

done

File: demos/primitive-mesh/index.ts (hunk 5 above)
Comment (singlecoder) [reply demos/primitive-mesh/index.ts]: > can remove setScale and let camera more near

done

File: demos/primitive-mesh/index.ts [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+cameraEntity.addComponent(Camera);
+cameraEntity.addComponent(OrbitControl);
+
+// Create direct light
+const lightEntity = rootEntity.createChild("DirectLight");
+const light = lightEntity.addComponent(DirectLight);
+light.color.setValue(1, 1, 1, 1);
+light.intensity = 0.3;
+
+function generatePrimitiveEntity(type: string, position: Vector3, material: Material, mesh: Mesh) {
```
Comment (GuoLei1990) [inline demos/primitive-mesh/index.ts:49]: write return type

Comment (GuoLei1990) [inline demos/primitive-mesh/index.ts:49]: use "name: string" is better than  "type: string"

**Review by gz65555 — APPROVED** [review]

**Pull request merged**